src/EnergyPlus/BranchNodeConnections.cc

```diff
@@ -1672,86 +1674,123 @@ void GetChildrenData(EnergyPlusData &state,
-                                                                    state.dataLoopNodes->NodeID({1, state.dataLoopNodes->NumOfNodes}),
-                                                                    state.dataLoopNodes->NumOfNodes);
-                    ChildOutNodeNum(CountNum) = Util::FindItemInList(ChildOutNodeName(CountNum),
-                                                                     state.dataLoopNodes->NodeID({1, state.dataLoopNodes->NumOfNodes}),
-                                                                     state.dataLoopNodes->NumOfNodes);
+                    ChildrenCType(CountNum) = ChildCType(MatchInNodeIndex);
+                    ChildrenCName(CountNum) = ChildCName(MatchInNodeIndex);
+                    InletNodeName(CountNum) = ChildInNodeName(MatchInNodeIndex);
+                    InletNodeNum(CountNum) = ChildInNodeNum(MatchInNodeIndex);
+                    OutletNodeName(CountNum) = ChildOutNodeName(MatchInNodeIndex);
+                    OutletNodeNum(CountNum) = ChildOutNodeNum(MatchInNodeIndex);
+                    ChildMatched(MatchInNodeIndex) = true;
+                    MatchInNodeNum = ChildOutNodeNum(MatchInNodeIndex);
                 }
-            }
-            if (CountNum != NumChildren) {
-                ShowSevereError(state, "GetChildrenData: Counted nodes not equal to GetNumChildren count");
-                ErrInObject = true;
-            } else {
-                // Children arrays built.  Now "sort" for flow connection order(?)
-                std::string MatchNodeName = ParentInletNodeName;
-                CountNum = 0;
-                int CountMatchLoop = 0;
-                while (CountMatchLoop < NumChildren) {
-                    ++CountMatchLoop;
-                    //          Matched=.FALSE.
-                    for (int Loop = 1; Loop <= NumChildren; ++Loop) {
-                        if (ChildInNodeName(Loop) == MatchNodeName) {
-                            ++CountNum;
-                            ChildrenCType(CountNum) = ChildCType(Loop);
-                            ChildrenCName(CountNum) = ChildCName(Loop);
-                            InletNodeName(CountNum) = ChildInNodeName(Loop);
-                            InletNodeNum(CountNum) = ChildInNodeNum(Loop);
-                            OutletNodeName(CountNum) = ChildOutNodeName(Loop);
-                            OutletNodeNum(CountNum) = ChildOutNodeNum(Loop);
-                            ChildInNodeName(Loop).clear(); // So it won't match anymore
-                            //              Matched=.TRUE.
-                            MatchNodeName = ChildOutNodeName(Loop);
-                            break;
-                        }
-                    }
+
+                for (int Loop = 1; Loop <= NumChildren; ++Loop) {
+                    if (ChildMatched(Loop)) continue;
+                    ++CountNum;
+                    ChildrenCType(CountNum) = ChildCType(Loop);
```

Comment:
Spot checked random set of files that were showing diffs. VSHeatPumpWaterHeater with debug build fails here with a vector error, `CountNum=5` but `NumChildren=4`.

Reply:
Hmmm. Why is this not showing up in the CI report? Or is it and I just don't know where to look?

Reply:
So this particular case crashes because there is a circularity in the node definitions:

- ParentInNodeName = "HPWHWATERINLETNODE"
- Component 2 "HPWHDXCOILVS", InNode = "HPWHWATERINLETNODE", OutNode = "HPWATEROUTLETNODE"
- Component 4 "HPWHTANK", InNode = "HPWATEROUTLETNODE", OutNode = "HPWHWATERINLETNODE"

The old code dealt with this issue by basically deleting sub-components from the temporary list if they have already been processed such that they could not be found again, and I could do something similar here. I assume this is a valid node configuration ...

Reply:
Yes, valid, intentional, as designed. The coil water nodes connect directly to the WH tank inlet/outlet nodes.

Reply:
> Hmmm. Why is this not showing up in the CI report? Or is it and I just don't know where to look?

It didn't get caught in release builds on gha, and it looks like the linux CI tests are not running today. @Myoldmopar ?
But I'm not 100% certain it would have failed there either.